### PR TITLE
LIT-197 Make fragment table size configureable

### PR DESCRIPTION
**Pull request by expressvpn-tom-l**
## Description
Added a new API to allow implementation customise the fragmentation table size.

## Motivation and Context
Reduce memory usage on certain devices.

## How Has This Been Tested?
Unit tested.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All active GitHub checks are passing  
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

File: src/he/conn.c [hunk 1]
```diff
@@ -198,6 +190,9 @@ he_return_code_t he_internal_conn_configure(he_conn_t *conn, he_ssl_ctx_t *ctx)
   // Initialize internal variables
   conn->ping_next_id = 1;
 
+  // Initialize the fragment table
+  conn->frag_table = he_internal_fragment_table_create(ctx->max_frag_entries);
```
Comment (expressvpn-raihaan-m) [inline src/he/conn.c:194]: If this fails should we error out?

File: src/he/frag.c [hunk 2]
```diff
@@ -184,3 +184,68 @@ he_return_code_t he_fragment_entry_update(he_fragment_entry_t *entry, uint8_t *d
...
+    tbl->entries[idx] = NULL;
+  }
+}
+
+void he_internal_fragment_table_destroy(he_fragment_table_t *tbl) {
+  if(!tbl) {
+    return;
+  }
+  // Free up all cached fragments
+  for(size_t i = 0; i < (sizeof(tbl->entries) / sizeof(he_fragment_entry_t *)); i++) {
```
Comment (expressvpn-raihaan-m) [inline src/he/frag.c:241]: this should be `i < tbl->num_entries`

File: src/he/conn.c (hunk 1 above)
Comment (expressvpn-tom-l) [reply src/he/conn.c:194]: Fixed.

File: src/he/frag.c (hunk 2 above)
Comment (expressvpn-tom-l) [reply src/he/frag.c]: Fixed.

**Review by expressvpn-raihaan-m — APPROVED** [review]

**Pull request merged**